### PR TITLE
WIP: Add assert for unknown OSR points

### PR DESCRIPTION
**Pull request by ncough**
This change explicitly adds checks for OSR points that are calls,
as in the past it would default to a call if no other type of
point was found. It will now default to preExecutionOSR behaviour
and assert.

### CODE REVIEW

File: compiler/compile/OMRCompilation.cpp [hunk 1]
```diff
@@ -787,11 +787,16 @@ OMR::Compilation::getOSRInductionOffset(TR::Node *node)
    if (self()->getOSRTransitionTarget() != TR::postExecutionOSR)
       return 0;
 
+   if (node->getOpCodeValue() == TR::treetop || node->getOpCode().isCheck())
```
Comment (vijaysun-omr) [inline compiler/compile/OMRCompilation.cpp]: "This change explicitly adds checks for OSR points that are calls"....um, where is the explicit check for a call ? I only see us checking the treetop node for opcodes that would be above calls.

Comment (ncough) [inline compiler/compile/OMRCompilation.cpp]: Throughout OSR functions an assumption is made that a treetop or check will have a call as a child, as this is the only treetop/check that can pass isPotentialOSRPoint. As this was the same assumption that resulted in the defaults this commit is attempting to address, I will add the check for calls as well.

**Comment by andrewcraik** [comment]
I'm not sure if you really want isCall here - there are a number of nodes that respond yes to isCall that are not yield points... At the very least some helpers don't need to be included nor do arraycopys for example.

**Comment by mstoodle** [comment]
@ncough Is this change still progressing? If not, could we close this PR and get you to submit a new one once the change is ready?

**Pull request closed without merging**